rust/main/chains/hyperlane-cosmos/src/providers/rpc/client.rs

```diff
@@ -132,6 +145,27 @@ impl CosmosRpcClient {
     }
 }
 
+impl Drop for CosmosRpcClient {
+    fn drop(&mut self) {
+        // decrement provider metric count
+        let chain_name = PrometheusConfig::chain_name(&self.metrics_config.chain);
+        self.metrics.decrement_provider_instance(chain_name);
+    }
+}
+
+impl Clone for CosmosRpcClient {
+    fn clone(&self) -> Self {
+        // increment provider metric count
+        let chain_name = PrometheusConfig::chain_name(&self.metrics_config.chain);
+        self.metrics.increment_provider_instance(chain_name);
+        Self {
+            client: self.client.clone(),
+            metrics: self.metrics.clone(),
+            metrics_config: self.metrics_config.clone(),
+        }
```

Comment:
now that we have the `new` method, this should be
```suggestion
        Self::new(
            client: self.client.clone(),
            metrics: self.metrics.clone(),
            metrics_config: self.metrics_config.clone(),
        )
```